ObjectiveGit/GTRepository.h

```diff
@@ -209,6 +231,18 @@ typedef NS_ENUM(NSInteger, GTRepositoryStateType) {
 /// Returns the initialized repository, or nil if an error occurred.
 - (nullable instancetype)initWithURL:(NSURL *)localFileURL error:(NSError **)error;
 
+/// Convenience initializer to find and open a repository with extended controls.
+///
+/// localFileURL - The file URL for the new repository. Cannot be nil.
+/// flags        - A combination of the `GTRepositoryOpenFlags` flags.
+/// ceilingDirs  - A GIT_PATH_LIST_SEPARATOR delimited list of path prefixes at
+///                which the search for a containing repository should terminate.
+///                Can be NULL.
+/// error        - The error if one occurs.
+///
+/// Returns the initialized repository, or nil if an error occurred.
+- (nullable instancetype)initWithURL:(NSURL *)localFileURL flags:(UInt32)flags ceilingDirs:(nullable const char *)ceilingDirs error:(NSError **)error;
```

Comment:
Usually, my Cocoa-fu says that convenience initializers get declared as class methods, designated initializers go under `-init...`. Arguably, this one could be kept as a designated initializer (since it ends up calling one of the handful `libgit2` functions that creates objects), but our designated initializer is actually `initWithGitRepository:`.

I'm not sure how that would like after being Swift-preprocessed though.